### PR TITLE
[Merged by Bors] - chore: cleanup of `mergeSort'`

### PR DESCRIPTION
**Pull request by kim-em**
This PR removes the old `mergeSort'` in Mathlib (which was *not* a stable sort algorithm), replacing all uses with the stable `mergeSort` from Lean, mostly preserving the API.

### CODE REVIEW

File: Mathlib/Data/List/Sort.lean [hunk 1]
```diff
@@ -378,210 +377,89 @@ end Correctness
...
+    rw [map_take, map_drop]
+    simp
   termination_by length l
 
 end MergeSort
 
 end sort
 
--- try them out!
---#eval insertionSort (fun m n : ℕ => m ≤ n) [5, 27, 221, 95, 17, 43, 7, 2, 98, 567, 23, 12]
```
Comment (vihdzp) [inline Mathlib/Data/List/Sort.lean:585]: I thought these comments were a cute demonstration of computability, I wouldn't want to see them go.

Comment (semorrison) [reply Mathlib/Data/List/Sort.lean:585]: I disagree?

Comment (jcommelin) [reply Mathlib/Data/List/Sort.lean:669]: Can we turn this into an `example` instead?
We have more of such examples in mathlib...

**Review by jcommelin — COMMENTED** [review]
bors d+

File: Mathlib/Data/List/Sort.lean (hunk 1 above)
Comment (kim-em) [reply Mathlib/Data/List/Sort.lean:669]:
```
unseal merge mergeSort in
example :
    mergeSort [5, 27, 221, 95, 17, 43, 7, 2, 98, 567, 23, 12] (fun m n => m / 10 ≤ n / 10) =
      [5, 7, 2, 17, 12, 27, 23, 43, 95, 98, 221, 567] := rfl
```      
One needs some `unseal`ing here to run the example by `rfl`.

**Comment by kim-em** [comment]
bors merge

**Comment by kim-em** [comment]
bors merge

**Comment by b-mehta** [comment]
Ouch, is there a reason these were deleted without deprecations? I just had a bunch of code break because of this, and it took quite a while to fix. I can't find a replacement for `sorted_mergeSort'`, which takes in an explicit relation to Prop and uses IsTotal and IsTrans typeclasses either...

**Pull request closed without merging**